### DISCUSSION TITLE
`set paste` has been deprecated, and what is the substitutes?

### DISCUSSION THREAD

**Discussion by nullptr-yxw** [neovim/neovim#27828] [Q&A]
As is mentioned in [the document](https://neovim.io/doc/user/deprecated.html#_options), `set paste` has been deprecated, so how can we implement a similar feature?

**Comment by justinmk** [comment]
The doc that you linked tries to explain it very clearly:

> Just Paste It.™ The ['paste'](https://neovim.io/doc/user/deprecated.html#'paste') option is obsolete: [paste](https://neovim.io/doc/user/provider.html#paste) is handled automatically when you paste text using your terminal's or GUI's paste feature (CTRL-SHIFT-v, CMD-v (macOS), middle-click, …).

'paste' is obsolete because you can just use your OS paste feature to paste things.
- If you're using a terminal that means `ctrl-v` or `ctrl-shift-v` or `cmd-v` (macOS). Or use your terminal's "Edit" menu and choose "Paste".
    - Middle-click also works, depending on your terminal.
- If you're using a Nvim GUI then the above should also work, though it may have its own menu or hotkeys.

**Reply by Deeperand** [reply]
hi! I found my paste cannot be correctly treated. e.g. when I copy the following code to clipboard (I'm using macOS with terminal iTerm)

```
int main()
{
  double x = 0.0;
  for (size_t i = 0; i < imax ; ++i)
  {
    x += i;
  }
}
```

then paste into NeoVim with `<cmd-v>`, it becomes

```
nt main()
{
  double x = 0.0;
    for (size_t i = 0; i < imax ; ++i)
      {
          x += i;
            }
            }

      }
}
```

I do understand this may lead by some auto-indent settings. But with `:set paste`, ALL behavior is as expected. So it looks automatically deal paste cannot completely replace `:set paste` (at least in my environment). Or if I need some extra settings to enable auto-paste mechanism working?

**Reply by justinmk** [reply]
> then paste into NeoVim with `<cmd-v>`, it becomes

Sounds like you have mapped `<d-v>`, because bracketed paste should not behave like that. Try using the Edit > Paste menu of iTerm, does it behave differently?
